spec/unit/library/helper_spec.rb

```diff
@@ -16,19 +16,77 @@
 #
 
 require 'rspec_helper'
+require 'ostruct'
 
-describe 'chomp_eol method' do
+describe 'helper methods' do
   before(:each) do
     @method_test = Class.new
     @method_test.extend(Line::Helper)
-    allow(@method_test).to receive(:new_resource).and_return(double('new_resource', eol: "\n"))
+    @method_test.extend(DSLHelpers)
+    new_resource = OpenStruct.new
+    new_resource.eol = "\n"
+    allow(@method_test).to receive(:new_resource).and_return(new_resource)
   end
 
-  it 'should remove trailing eol' do
-    expect(@method_test.chomp_eol("line\n")).to eq('line')
+  describe 'chomp_eol method' do
+    it 'should remove trailing eol' do
+      expect(@method_test.chomp_eol("line\n")).to eq('line')
+    end
+
+    it 'should raise error with embedded eol' do
+      expect { @method_test.chomp_eol("embedded\ninline") }.to raise_error(ArgumentError)
+    end
+  end
+
+  describe 'default_eol' do
+    it 'should return an eol character' do
+      expect(@method_test.default_eol).to eq("\n")
+    end
+  end
+
+  describe 'filters' do
+    it 'should return a new filter object' do
+      expect(@method_test.filters.class).to eq(Line::Filter)
+    end
+  end
+
+  describe 'target_current_lines?' do
+    it 'should get an array of lines' do
+      File.write('/tmp/file', "foo\nbar\nlast\n")
+      new_resource = OpenStruct.new
+      new_resource.eol = "\n"
+      new_resource.path = '/tmp/file'
+      allow(@method_test).to receive(:new_resource).and_return(new_resource)
+      expect(@method_test.target_current_lines).to eq(%w(foo bar last))
+    end
+  end
+
+  describe 'target_file_exist?' do
+    it 'should verify file does not exist' do
+      new_resource = OpenStruct.new
+      new_resource.eol = "\n"
+      new_resource.path = '/tmp/nofile'
+      allow(@method_test).to receive(:new_resource).and_return(new_resource)
+      expect(@method_test.target_file_exist?).to eq(false)
+    end
+    it 'should verify file does exist' do
+      File.write('/tmp/file', 'foo')
+      new_resource = OpenStruct.new
+      new_resource.eol = "\n"
+      new_resource.path = '/tmp/file'
+      allow(@method_test).to receive(:new_resource).and_return(new_resource)
+      expect(@method_test.target_file_exist?).to eq(true)
+    end
+  end
+end
+
+# Replace the chef DSL methods
+module DSLHelpers
+  def property_is_set?(_property)
```

Comment:
same question about this always being set to false?

Reply:
The only thing that's effected is the setting of sensitive and I can't really test that with an rspec tests.  Tests of the use of sensitive are in the test kitchen tests.